libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -514,14 +514,11 @@ void Compass::_detect_backends(void)
                      AP_Compass_LSM303D::name, false);
     }
     if (AP_BoardConfig::get_board_type() == AP_BoardConfig::PX4_BOARD_TEST_V3) {
-        _add_backend(AP_Compass_HMC5843::probe(*this, hal.spi->get_device(HAL_COMPASS_HMC5843_NAME),
```

Comment:
thats the SPI compass - I've moved the detection of it to the individual boards as I found some boards had it with a different orientation